datafusion/functions/src/regex/regexpcount.rs

```diff
@@ -343,7 +342,7 @@ where
                     .map(|(value, flags)| {
                         let pattern =
                             compile_and_cache_regex(regex, flags, &mut regex_cache)?;
-                        count_matches(value, &pattern, start_scalar)
+                        count_matches(value, pattern, start_scalar)
```

Comment:
I don't get why it even needs to cache a scalar regex? I think the compilation can be done earlier, so it can skip the lookup / cache and just reuse the compiled regex.

Reply:
Ah nvm, the flag is non-scalar in this example